### PR TITLE
Powheg-bb4l FSR veto

### PR DESCRIPTION
**Pull request by mseidel42**
FSR emission veto for Powheg-bb4l by Tomas Jezo

71X: #18756

### CODE REVIEW

**Comment by perrozzi** [comment]
please test

**Comment by perrozzi** [comment]
+1

File: GeneratorInterface/Pythia8Interface/plugins/PowhegHooksBB4L.h [hunk 1]
```diff
@@ -0,0 +1,330 @@
+// PowhegHooksBB4L.h 
+// Copyright (C) 2017 Tomas Jezo, Markus Seidel, Ben Nachman
```
Comment (davidlange6) [inline GeneratorInterface/Pythia8Interface/plugins/PowhegHooksBB4L.h]: please remove this.

File: GeneratorInterface/Pythia8Interface/plugins/PowhegHooksBB4L.h [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+		if (nDau == 0) {
+			// No resonance found, set scale to high value
+			// Pythia will shower any MC generated resonance unrestricted
+			scale = 1e30;
+		}
+		else if (nDau < 3) {
+			// No radiating resonance found
+			scale = 0.8;
+		}
+		else if (abs(event[iRes].id()) == 6) {
```
Comment (davidlange6) [inline GeneratorInterface/Pythia8Interface/plugins/PowhegHooksBB4L.h]: hi @intrepid42 - please use std::abs

File: GeneratorInterface/Pythia8Interface/plugins/PowhegHooksBB4L.h [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+
+	inline double gSplittingScale(Vec4 pt, Vec4 p1, Vec4 p2){
+		p1.bstback(pt);
+		p2.bstback(pt);		
+		return sqrt( 2*p1*p2*p1.e()*p2.e()/(pow(p1.e(),2)+pow(p2.e(),2)) );
+	}
+
+
+	inline double getdechardness(int topcharge, const Event &e){
+		int tid = 6*topcharge, wid = 24*topcharge, bid = 5*topcharge, gid = 21, wildcard = 0;
```
Comment (davidlange6) [inline GeneratorInterface/Pythia8Interface/plugins/PowhegHooksBB4L.h]: please add a comment to explain these magic numbers

**Comment by intrepid42** [comment]
Thanks for your comments @davidlange6 

I will backport to 71X (#18756) once this is approved.

**Comment by davidlange6** [comment]
please test

**Comment by davidlange6** [comment]
hi @intrepid42 - thanks - could you also fix the other ~5 abs -> std::abs. Thanks

**Comment by intrepid42** [comment]
@davidlange6 Ah, sure, sorry for that

**Comment by davidlange6** [comment]
please test

**Comment by perrozzi** [comment]
+1

**Comment by davidlange6** [comment]
+1

**Pull request merged**